forms-flow-web/src/components/AccessDenied/index.js

```diff
@@ -3,20 +3,36 @@ import { kcServiceInstance } from "../PrivateRoute"; // Import the kcServiceInst
 import { ReactComponent as AccessDeniedIcon } from "./AccessDenied.svg";
 import './accessDenied.scss';
 import { useTranslation } from "react-i18next";
+import { BASE_ROUTE } from "../../constants/constants";
+import { useHistory } from "react-router-dom";
+import { useSelector } from "react-redux";
+
+
 const AccessDenied = () => {
   const { t } = useTranslation();
+  const history = useHistory();
+  const userRoles = useSelector((state) => state.user.roles);
+
   const handleLogout = () => {
     const kcInstance = kcServiceInstance(); // Get the Keycloak instance
     kcInstance.userLogout();
   };
 
+  const handleReturn = () => {
+    history.push(BASE_ROUTE);
+  };
+
   return (
       <div className="d-flex flex-column  align-items-center text-center">
           <AccessDeniedIcon alt="Access Denied Icon" className="mb-4 mt-2"/>
       <h1 className="access-denied-text">{t("Access Denied")}</h1>
       <span className="access-denied">{t("You don't have permission to access this page.")}</span>
       <span className="access-denied">{t("Please contact your administrator or try again later.")}</span>
-      <button className="btn btn-primary me-1 mt-4" onClick={handleLogout}>{t("Return to login")}</button>
+      {userRoles.length === 0 ? (

```

Comment:
believe we can keep this component isolated and not dependent on specific user scenarios. This would allow us to reuse the component in different contexts. To achieve this, we should add conditional checks. For instance, if returnToHomeButton is true, we can display the button. Similarly, we can introduce other options based on different conditions. @arun-s-aot @Bonymol-aot

Reply:
Thanks for the  update Bony. As discussed on call , please pass userRoles alone to <AccessDenied> component and set the variables like showReturnToLogin & showReturnToHome from within that component

Reply:
Updated the changes 👍